[PVR][Estuary] Add missing EpisodePart information

emveepee opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
PVR allows storing data for a shows that are broken into parts (e.g. 2/5) , however this data is only managed internally.  This exposes the date to skins in the EPISODE tag.
<!--- Describe your change in detail here. -->

If the EpisodePart value is set and greater then zero (zero is valid in PVR but makes no sense, probably bad XMLTV parsing)  a formatted string is created from the EpisodeNumber and EpisodePart 

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

This useful data is not exposed to users.  Since the logic and database structure is already in place  this doesn't make sense

A future PR might address the poorly named field EpisodePart   it is TotalParts

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Tested on Windows with pvr.nextpvr.  Typically the number of parts will only a small number of characters to the display so should only impact skin display region.  Screenshots for Estuary indicate room is available.  Often these  shows won't have series information so already three characters shorter they serial shows.

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

More  information for users.

## Screenshots (if appropriate):

These are for the 2 shows one from the EPG and the following on the dialog.  

![image](https://github.com/xbmc/xbmc/assets/2148031/b0f209f3-d3e0-4677-a2f1-d21d732acddf)

![image](https://github.com/xbmc/xbmc/assets/2148031/4d00512f-9201-40de-b5fc-456dcf8b2ca4)

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x ] All new and existing tests passed

emveepee committed:
[PVR] Add missing EpisodePart information

PVR allows storing data for a shows that are broken into parts (e.g. 2/5) , however this data is only managed internally.  This exposes the data to skins adding two new Infolabel  ListItem.EpisodePart and  VideoPlayer.EpisodePart

ksooo commented on xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp:
```diff
@@ -775,7 +775,11 @@ bool CPVRGUIInfo::GetListItemAndPlayerLabel(const CFileItem* item,
-          strValue = std::to_string(epgTag->EpisodeNumber());
+          if (epgTag->EpisodePart() > 0)
+            strValue =
+                StringUtils::Format("{0}/{1}", epgTag->EpisodeNumber(), epgTag->EpisodePart());
```
Are we sure that in all languages the separator between episode number and episode part is a forward slash or should this rather be a translatable string making it possible to translate the separator?

emveepee replied:
Fair question however if you can consider it fractional math I assumed the divisor was universal.  If you want the change I guess it would be a global language resource not skin.

Would this mean that the S and E need to be changed too?

ksooo replied:
> Would this mean that the S and E need to be changed too?

Strictly speaking, yes. For example in German “episode” can be translated to “Episode”, but this is not a good translation. Best would be “Folge” but if we would use this for Kodi, due to the hard coded E “everywhere”, we would get inconsistent. Netflix for example uses “F” and “Folge”.  They do it right.

emveepee commented:
@ksooo I added a second commit as a test this PR.  Here is the timer skin from pvr.nextpvr  the duplicated information makes the list hard to follow.  If there had been an EpisodeName that could easily be truncated.  If you don't think the PR's are related I can resubmit. it.

Before

![image](https://github.com/xbmc/xbmc/assets/2148031/722bcb4e-9058-40b5-a520-d2a0bfeb986c)

After

![image](https://github.com/xbmc/xbmc/assets/2148031/422cb166-ed41-46bc-955f-33249d581055)

ksooo commented:
I don’t want to be the party pooper, but imo extending LISTITEM_EPISODE is the wrong approach. Introduction of a new GUI label carrying the episode part and extending skins to use it is the way to go. The skins should should decide where and how to visualize episode number and episode part. 

Changing LISTITEM_EPISODE is the easiest approach but can lead to multiple problems, starting with now to small labels to display the info in some screen but more problematic no longer having a GUI label which is guaranteed to return only the episode number in case a skinner wants exactly this information and not the episode part.

ksooo commented:
> If you don't think the PR's are related I can resubmit.

I consider this a nice enhancement, but interrogated to this PR. Please resubmit.

emveepee commented:
> I don’t want to be the party pooper, but imo extending LISTITEM_EPISODE is the wrong approach. Introduction of a new GUI label carrying the episode part and extending skins to use it is the way to go. The skins should should decide where and how to visualize episode number and episode part.
> 
> Changing LISTITEM_EPISODE is the easiest approach but can lead to multiple problems, starting with now to small labels to display the info in some skins but more problematic no longer having a GUI label which is guaranteed to return only the episode number in case a skinner wants to display exactly this information and not the episode part.

No problem, I was hoping to expose this data in Omega but I guess it has been missing for a while.  Screenshots in first post are with the new logic.

ksooo commented:
For completeness, we also need VIDEOPLAYER_EPISODEPART, to be able to show this information in the video OSD. 
Not sure, but if we already show season/episode info in estuary video OSD, the episode part should be added in this PR as well.

emveepee commented:
> For completeness, we also need VIDEOPLAYER_EPISODEPART, to be able to show this information in the video OSD. Not sure, but if we already show season/episode info in estuary video OSD, the episode part should be added in this PR as well.

@ksooo updated and tested with Live TV.  Recording don't have an episodepart though that will take an API version bump (I will attempt that next).  I also remove a lingering " - " if no episode name is available.

I will need assistance from @Hitcher or @jjd-uk on how to test PlayerLabel2  I couldn't seem to trigger it.

ksooo commented:
> Recording don't have an episodepart though that will take an API version bump

Despite of the pvr addon api bump for the nrw field for PVR_RECORDING, you need to extend CVideoInfoTag, not CPVRRecoRding please. So, every video could provide episode part information, not only recordings.

emveepee commented:
> > Recording don't have an episodepart though that will take an API version bump
> 
> Despite of the PVR addon API bump for the new field for PVR_RECORDING, you need to extend CVideoInfoTag, not CPVRRecording please. So, every video could provide episode part information, not only recordings. Thus, implementation for the GUI info label for recordings utilizing the new CVideoInfoTag field belongs to VideoGUIInfo.cpp, not PVRGUIInfo.cpp.

It is working from Live TV without now without changing CVideoInfoTag is that necessary?  I think that should wait to match the parsers and database changes required and I am not sure that TMDB or TVDB even provide this.

ksooo commented:
If you want to provide support for recordings, the way to go is to extend CVideoInfoTag. CPVRRecording is derived from it and the fields for episode num etc come from CVideoInfoTag. No problem, if as a first step only pvr recordings and no normal videos support episode part.

emveepee commented:
> If you want to provide support for recordings, the way to go is to extend CVideoInfoTag. CPVRRecording is derived from it and the fields for episode num etc come from CVideoInfoTag. No problem, if as a first step only pvr recordings and no normal videos support episode part.

@ksooo OK if you and the Estuary reviewers can review this as a standalone PR without the video tag support I will research research and code PVR recording.  If it is able to be merged, I will then sync the recording PR @DeltaMikeCharlie API bump 

For recordings honestly it might be just easier to assume season 1, since that is what TMBD and TVDB (search a current show "Night Caller")  as an example.

ksooo commented:
Fine for me to add support for recordings in a separate PR. Let’s do this step by step.

jjd-uk commented:
Just a quick ack to say I've seen the ping, I don't have a great deal of free time but I'll try and take a look over the weekend.

DeltaMikeCharlie commented on xbmc/guilib/guiinfo/GUIInfoLabels.h:
```diff
@@ -989,6 +990,7 @@ static constexpr unsigned int SYSTEM_LOCALE = 1012;
 #define LISTITEM_PVR_INSTANCE_NAME  (LISTITEM_START + 218)
 #define LISTITEM_CHANNEL_LOGO       (LISTITEM_START + 219)
 #define LISTITEM_PVR_GROUP_ORIGIN   (LISTITEM_START + 220)
+#define LISTITEM_EPISODEPART        (LISTITEM_START + 221)
```
This value (LISTITEM_START + 221) will collide with unmerged PR#24096.

https://github.com/xbmc/xbmc/blob/affc44682761fbcbc9243b840c91d6780be8301c/xbmc/guilib/guiinfo/GUIInfoLabels.h#L992C1-L992C61

ksooo replied:
Whichever PR gets merged first, the other PR needs to be rebased to resolve the conflict. Nothing that needs to be charged in this PR as if now.

ksooo commented on xbmc/interfaces/json-rpc/schema/types.json:
```diff
@@ -3881,6 +3881,7 @@
         "firstaired",
         "season",
         "episode",
+        "episodepart",
```
This change either has to go into your other PR (which I suggest) or json-rpc API needs to be bumped in this PR.

emveepee replied:
OK I consider that part of the "missing" EpisodePart information but that is fine they are pretty tightly coupled.

emveepee commented on xbmc/GUIInfoManager.cpp:
```diff
@@ -3954,6 +3954,15 @@ const infomap musicplayer[] =    {{ "title",            MUSICPLAYER_TITLE },
+///     @return string containing the number of parts of a single episode - empty if no data provided
+///     <p><hr>
+///     @skinning_v22 **[Infolabel Updated]** \link VideoPlayer_EpisodePart `VideoPlayer.EpisodePart`\endlink
+///     also supports EPG.
```
It seemed like a comment to skinners, I don't know really know what it means (maybe PVR) but it is also on

```
///     @skinning_v15 **[Infolabel Updated]** \link VideoPlayer_Episode `VideoPlayer.Season`\endlink
///     also supports EPG.
...
///     @skinning_v15 **[Infolabel Updated]** \link VideoPlayer_Director `VideoPlayer.Episode`\endlink
///     also supports EPG.
```
so I wanted consistency.

ksooo reviewed (approved):
C++ changes look okay to me.

jjd-uk commented:
That Estuary Variable changes look to be sensible however it appears I have no way to test since my UK TV EPG from TVHeadend does not seem to include this info.

> I will need assistance from @Hitcher or @jjd-uk on how to test PlayerLabel2 I couldn't seem to trigger it.

My TVHeadend client doesn't seem to provide any season or episode numbering via either ListItem.Season/ListItem.Episode or VideoPlayer.Season/VideoPlayer.Episode so I've never seen it trigger on real content. I assume it does trigger when provided info as I see it via Demo Client

![image](https://github.com/user-attachments/assets/4bcf3f54-5c64-4c5b-8064-c99e9c3bf330)

emveepee commented:
@jjd-uk   pvr.demo XML file I added here https://forum.kodi.tv/showthread.php?tid=378181&pid=3203924#pid3203924 has it on channel 8.  I thought it would be test how it looks for x based skins but the skins have to change (that precipitated the whole discussion)

Otherwise I think this may only be in pvr.iptvsimple and pvr.vuplus right now, I have actually just added it for pvr.nextpvr after only a few user's requests.  It is funny the work put into Kodi for it even a db field and no way to to use it.  I was first thinking of asking @ksooo if it should be deprecated but that is more work for core and all addons.

If you could send me your EPG16.db I could tell you if you should be seeing it.  Otherwise search the description of "Paddy & Molly", "Garden Rescue" and "Come Dine with Me" and see if they have what looks like a part number .  "Come Dine with me is interesting in that it has been on for many years, but the episodes are based on the region and year. If you follow the link https://forum.kodi.tv/showthread.php?tid=378181&pid=3203924#pid3203924 you can see it in the description. It might look different if you are using the OpenSky or XMLTV guide data.

Finally if you want to install pvr.nextpvr with OTA data the new addon is on jenkins.  I wanted to have a nightly for a user to test before merging it.

emveepee commented:
@jjd-uk you need more than the XML file I added HOTD to the XML file from this PR https://github.com/kodi-pvr/pvr.demo/pull/155 but it doesn't get built by jenkins so someone would need to trigger it.

jjd-uk commented:
Ok I'll try with that PVRDemoAddonSettings.xml file for Demo Client. My EPG setup on TVHeadend is just the standard 7 day epg embedded into the over air UK Freeview service.

DeltaMikeCharlie commented:
> My TVHeadend client doesn't seem to provide any season or episode numbering

@jjd-uk - I had a quick look at the TVH HTSP documentation.  The EPG message type seems to contain the season/episode data.

https://docs.tvheadend.org/documentation/development/htsp/server-to-client-methods#eventadd

However, these fields seem to be missing from the recording message type.

https://docs.tvheadend.org/documentation/development/htsp/server-to-client-methods#dvrentryadd

This was only a 'quick look' at the documentation, A more thorough search of the source code may uncover more.

emveepee commented:
> > My TVHeadend client doesn't seem to provide any season or episode numbering
> 
> @jjd-uk - I had a quick look at the TVH HTSP documentation. The EPG message type seems to contain the season/episode data.
> 
> https://docs.tvheadend.org/documentation/development/htsp/server-to-client-methods#eventadd

For this PR it looks we are talking about partNumber and partCount 

Kodi doesn't support EpisodePart (TVH partCount) on recordings so that is why I need to work on an API bump.

ksooo commented:
> so that is why I need to work on an API bump PR in Kodi.

Correct. Let's not start scope creep for this PR. We decided to handle recordings later, in another PR.

All we need for this PR is approval for the skin changes from @jjd-uk and than it can go in.

jjd-uk commented:
I found a Windows artifact for pvr.demo and with the xml from the thread I've now been able to test, and as a consequence I can see some Estuary changes will be necessary.

jjd-uk commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -416,7 +416,10 @@
+		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodePart)">$INFO[VideoPlayer.Season,[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906][/CAPITALIZE]:[/COLOR] S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.EpisodePart,/]</value>
+		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodeName)">$INFO[VideoPlayer.Season,[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906][/CAPITALIZE]:[/COLOR] S]$INFO[VideoPlayer.Episode,E, - ]$INFO[VideoPlayer.EpisodeName]</value>
+		<value condition="VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag">$INFO[VideoPlayer.Season,[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906][/CAPITALIZE]:[/COLOR] S]$INFO[VideoPlayer.Episode,E]</value>
 		<value condition="player.chaptercount + [!VideoPlayer.Content(episodes) + !VideoPlayer.Content(LiveTV)]">$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
```
Can you change `OSDSubLabelVar` to:
```
	<variable name="OSDSubLabelVar">
		<value condition="Window.IsActive(visualisation) + Integer.IsGreater(Playlist.Length(music),1) + Integer.IsGreater(Playlist.Position(music),0)">$LOCALIZE[554] $INFO[Playlist.Position] / $INFO[Playlist.Length]</value>
		<value condition="VideoPlayer.Content(musicvideos)">$VAR[NowPlayingSublabelVar,,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]: [/COLOR]]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
		<value condition="VideoPlayer.Content(episodes) + !player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E,: ]$INFO[VideoPlayer.Title]</value>
		<value condition="VideoPlayer.Content(episodes) + player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E, - ]$INFO[VideoPlayer.Title,,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodeName) + !String.IsEmpty(VideoPlayer.EpisodePart)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.EpisodePart,/, - ]$INFO[VideoPlayer.EpisodeName]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodePart)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.EpisodePart,/]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodeName)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E, - ]$INFO[VideoPlayer.EpisodeName]</value>
		<value condition="VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag + !String.IsEmpty(VideoPlayer.Episode)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]</value>
		<value condition="player.chaptercount + [!VideoPlayer.Content(episodes) + !VideoPlayer.Content(LiveTV)]">$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
		<value>$INFO[VideoPlayer.Genre]</value>
	</variable>
```

This will now give

**AFTER**
![image](https://github.com/user-attachments/assets/5f917587-9471-47cc-aa42-509c1b4c8e66)

Before the $LOCALIZE[36906] for Episode would only display if VideoPlayer.Season returned something, so before the change the string $LOCALIZE[36906] for Episode was not displayed using the example in pvr.demo.

**BEFORE**
![image](https://github.com/user-attachments/assets/0c316039-56d2-4d5d-a716-180990a5e35d)

emveepee replied:
Sure, it probaly wasn't there originally as you showed in your earlier screenshot, since when you have the localized word "Episode" the E is actually redundant.

jjd-uk replied:
Where there is both Season and Episode info then it's shown as:

![image](https://github.com/user-attachments/assets/9f38160e-c4af-4ca1-aa28-338187a39202)

However now I think of it more the you can probably drop the `E` so make it:

`<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodeName) + !String.IsEmpty(VideoPlayer.EpisodePart)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode]$INFO[VideoPlayer.EpisodePart,/, - ]$INFO[VideoPlayer.EpisodeName]</value>`

giving

![image](https://github.com/user-attachments/assets/fa9c4eba-1e8d-40b8-88d5-dcc99211047b)

As it the consistency of displaying the label $LOCALIZE[36906] for Episode I'm after.

emveepee replied:
> However now I think of it more the you can probably drop the `E` so make it:
> 
> `<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodeName) + !String.IsEmpty(VideoPlayer.EpisodePart)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode]$INFO[VideoPlayer.EpisodePart,/, - ]$INFO[VideoPlayer.EpisodeName]</value>`


In may last push I didn't include that (although it does look better) because it makes the assumption that the E isn't required when the episodepart is included.  For VideoPlayer that assumption should be when season is empty.

jjd-uk replied:
Having gone through all the styles of EPG entries in pvr.demo I've now arrived at, let me know if I've missed anything that needs to be accounted for.

```
	<variable name="OSDSubLabelVar">
		<value condition="Window.IsActive(visualisation) + Integer.IsGreater(Playlist.Length(music),1) + Integer.IsGreater(Playlist.Position(music),0)">$LOCALIZE[554] $INFO[Playlist.Position] / $INFO[Playlist.Length]</value>
		<value condition="VideoPlayer.Content(musicvideos)">$VAR[NowPlayingSublabelVar,,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]: [/COLOR]]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
		<value condition="VideoPlayer.Content(episodes) + !player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.Title, - ]</value>
		<value condition="VideoPlayer.Content(episodes) + player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.Title, - ,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodePart) + !String.IsEmpty(VideoPlayer.Season)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.EpisodePart,/]$INFO[VideoPlayer.EpisodeName, - ]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.EpisodePart) + String.IsEmpty(VideoPlayer.Season)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Episode]$INFO[VideoPlayer.EpisodePart,/]$INFO[VideoPlayer.EpisodeName, - ]</value>		
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag] + !String.IsEmpty(VideoPlayer.Season)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.EpisodeName, - ]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag] + String.IsEmpty(VideoPlayer.Season) + !String.IsEmpty(VideoPlayer.Episode)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Episode]$INFO[VideoPlayer.EpisodeName, - ]</value>
		<value condition="[VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag] + String.IsEmpty(VideoPlayer.Season) + String.IsEmpty(VideoPlayer.Episode)">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.EpisodeName]</value>
		<value condition="player.chaptercount + [!VideoPlayer.Content(episodes) + !VideoPlayer.Content(LiveTV)]">$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
		<value>$INFO[VideoPlayer.Genre]</value>
	</variable>
```

jjd-uk replied:
Ignore the new commit with `VideoPlayerSeasonEpisodeLabel` seems to encompass the same use cases.

jjd-uk commented:
> 
> I will need assistance from @Hitcher or @jjd-uk on how to test PlayerLabel2 I couldn't seem to trigger it.

I know understand what you meant by this. Variable `PlayerLabel2` is used in the player controls dialog at https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/xml/PlayerControls.xml#L34

Player Controls is actived while outside of fullscreen mode using keybaord shortcut Ctrl+n

![image](https://github.com/user-attachments/assets/532d516f-53c6-4cdf-afc4-c2c461abb113)

It looks like this will also need some attention.

jjd-uk commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -580,6 +583,7 @@
 	</variable>
 	<variable name="PlayerLabel2">
+		<value condition="VideoPlayer.Content(episodes) + !String.IsEmpty(VideoPlayer.EpisodePart)">$LOCALIZE[20373]$INFO[VideoPlayer.Season,: , / ]$LOCALIZE[20359]$INFO[VideoPlayer.Episode,/]$INFO[VideoPlayer.EpisodePart,: ]</value>
 		<value condition="VideoPlayer.Content(episodes)">$LOCALIZE[20373]$INFO[VideoPlayer.Season,: , / ]$LOCALIZE[20359]$INFO[VideoPlayer.Episode,: ]</value>
```
Remove this change to variable `PlayerLabel2` instead I'm going to suggest a change to variable `PlayerLabel3`

emveepee replied:
Sure I only put it there because seemed related. I still can't trigger it do you have a keymap for Ctrl-N?

jjd-uk replied:
No special keymap just the default one.

If you want to add it to your own key combo it's action `ActivateWindow(playercontrols)`

jjd-uk commented:
With variable `PlayerLabel2` removed, then change variable PlayerLabel3 here

https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/xml/Variables.xml#L588-L592

to:

```
	<variable name="PlayerLabel3">
		<value condition="VideoPlayer.Content(livetv) + !String.IsEmpty(VideoPlayer.Episode) + String.IsEmpty(VideoPlayer.EpisodePart)">$INFO[VideoPlayer.Season,$LOCALIZE[20373]: , ]$INFO[VideoPlayer.Episode,$LOCALIZE[20359]: ]</value>	
		<value condition="VideoPlayer.Content(livetv) + !String.IsEmpty(VideoPlayer.Episode) + !String.IsEmpty(VideoPlayer.EpisodePart)">$INFO[VideoPlayer.Season,$LOCALIZE[20373]: , ]$INFO[VideoPlayer.Episode,$LOCALIZE[20359]: ]$INFO[VideoPlayer.EpisodePart,/]</value>
		<value condition="VideoPlayer.Content(movies) | VideoPlayer.Content(livetv) + String.IsEmpty(VideoPlayer.Episode)">$INFO[VideoPlayer.Genre]</value>
		<value condition="VideoPlayer.Content(episodes)">$INFO[VideoPlayer.TvShowTitle]</value>
		<value condition="Player.HasAudio">$INFO[MusicPlayer.TrackNumber,,: ][COLOR=grey]$INFO[Player.Title][/COLOR]</value>
	</variable>
```

This gives:

![image](https://github.com/user-attachments/assets/5aba332b-dd17-44a4-942c-45cb6d47bbc0)

also adds missing Season and Episode info for **livetv**

![image](https://github.com/user-attachments/assets/35a09ef9-5fa7-44ab-997c-873de2b4299e)

emveepee commented:
> With variable `PlayerLabel2` removed, then change variable PlayerLabel3 here

Now it isn't consistent, shouldn't you either not prefix VideoPlayer with the E or orefix this with the E

jjd-uk commented:
> > With variable `PlayerLabel2` removed, then change variable PlayerLabel3 here
> 
> Now it isn't consistent, shouldn't you either not prefix VideoPlayer with the E or prefix this with the E

The aim was to make it consistent with the labelling for normal TV show episodes (non-LiveTV) as in https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/xml/Variables.xml#L583

emveepee committed:
Season episode changes

Additional changes to skin Episode labeling removing and VideoPlayer  removing leading E from @jjd-uk

emveepee commented:
@jjd-uk I added a second commit to try and consolidate all those episode variables and easier support for the VideoPlayer.  I think it encompasses your suggestion for VideoPlayer.    On the list I added highlites to the recording list, you may not agree.  The Full display dialog is no longer in bold just white to allow common functions.

jjd-uk commented:
> @jjd-uk I added a second commit to try and consolidate all those episode variables and easier support for the VideoPlayer. I think it encompasses your suggestion for VideoPlayer. On the list I added highlites to the recording list, you may not agree. The Full display dialog is no longer in bold just white to allow common functions.

Hadn't spotted this before making my last comment.

jjd-uk commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -414,9 +414,9 @@
-		<value condition="VideoPlayer.Content(episodes) + player.chaptercount">$INFO[VideoPlayer.Season,[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906][/CAPITALIZE]:[/COLOR] S]$INFO[VideoPlayer.Episode,E, - ]$INFO[VideoPlayer.Title,,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
-		<value condition="VideoPlayer.Content(LiveTV) | PVR.IsPlayingRecording | PVR.IsPlayingEpgTag">$INFO[VideoPlayer.Season,[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906][/CAPITALIZE]:[/COLOR] S]$INFO[VideoPlayer.Episode,E, - ]$INFO[VideoPlayer.EpisodeName]</value>
+		<value condition="VideoPlayer.Content(episodes) + !player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E,: ]$INFO[VideoPlayer.Title]</value>
+		<value condition="VideoPlayer.Content(episodes) + player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E, - ]$INFO[VideoPlayer.Title,,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterC ...
```
Apologies a couple lines here are not ideal for `VideoPlayer.Title` so need changing, probably my fault from one of my code snippets.

```
		<value condition="VideoPlayer.Content(episodes) + !player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.Title, - ]</value>
		<value condition="VideoPlayer.Content(episodes) + player.chaptercount + ![String.IsEmpty(VideoPlayer.Episode) | String.IsEmpty(VideoPlayer.Title)]">[COLOR button_focus][CAPITALIZE]$LOCALIZE[36906]:[/CAPITALIZE][/COLOR] $INFO[VideoPlayer.Season,S]$INFO[VideoPlayer.Episode,E]$INFO[VideoPlayer.Title, - ,[CR]]$INFO[player.chapter,[COLOR button_focus]$LOCALIZE[21396]:[/COLOR] ]$INFO[Player.ChapterCount,/]$INFO[Player.ChapterName, - ]</value>
```

emveepee replied:
OK changed that.  Should VideoPlayer.Content(episodes) follow x notation not S/E since they are being loaded from x based dialogs and lists?

emveepee commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -605,8 +607,17 @@
+	<variable name="SeasonHighliteEpisodeLabel">
+		<value condition="!String.IsEmpty(ListItem.Season) | !String.IsEmpty(ListItem.Episode)">[COLOR grey]$INFO[ListItem.Season,S]$INFO[ListItem.Episode,E]$INFO[ListItem.EpisodePart,/][/COLOR][COLOR white]$INFO[ListItem.EpisodeName,: ][/COLOR]</value>
+		<value condition="!String.IsEmpty(VideoPlayer.EpisodeName)">[COLOR white]$INFO[ListItem.EpisodeName][/COLOR]</value>
+	</variable>
```
I didn't like the name either I was explaining what it was doing.  I don't know much about skinning so thanks a lot for the tip and help.  I also made the semi-colon grey and took out these two conditions because (I think) the $INFO checks anyways
```
		<value condition="!String.IsEmpty(ListItem.EpisodeName)">$INFO[ListItem.EpisodeName]</value>
to 
		<value>$INFO[ListItem.EpisodeName]</value>
``
@ksooo do you want to to squash this to one commit?

jjd-uk reviewed (approved):
Thanks for your patience. After going over it one more time I think we're there now.

ksooo commented on addons/skin.estuary/xml/Home.xml:
```diff
@@ -509,7 +509,7 @@
 							<param name="list_id" value="13400"/>
 							<param name="label" value="$INFO[ListItem.Title]$INFO[ListItem.Date, (,)]"/>
-							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
+							<param name="label2" value="$VAR[SeasonEpisodeLabel]"/>
```
Should this read `$VAR[ListItemSeasonEpisodeLabel]` ?

ksooo commented on addons/skin.estuary/xml/Home.xml:
```diff
@@ -498,7 +498,7 @@
 							<param name="list_id" value="13300"/>
 							<param name="label" value="$INFO[ListItem.Title]$INFO[ListItem.Date, (,)]"/>
-							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
+							<param name="label2" value="$VAR[SeasonEpisodeLabel]"/>
```
Should this read `$VAR[ListItemSeasonEpisodeLabel]` ?

ksooo commented on addons/skin.estuary/xml/Home.xml:
```diff
@@ -426,7 +426,7 @@
 							<param name="list_id" value="12400"/>
 							<param name="label" value="$INFO[ListItem.Title]$INFO[ListItem.Date, (,)]"/>
-							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
+							<param name="label2" value="$VAR[SeasonEpisodeLabel]"/>
```
Should this read `$VAR[ListItemSeasonEpisodeLabel]` ?

ksooo commented on addons/skin.estuary/xml/Home.xml:
```diff
@@ -415,7 +415,7 @@
 							<param name="list_id" value="12300"/>
 							<param name="label" value="$INFO[ListItem.Title]$INFO[ListItem.Date, (,)]"/>
-							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
+							<param name="label2" value="$VAR[SeasonEpisodeLabel]"/>
```
Should this read `$VAR[ListItemSeasonEpisodeLabel]` ?

ksooo reviewed (commented):
1) Uncommented force-pushes with additional functional changes after approval is not a good approach.
2) Why do we have now `$VAR[ListItemSeasonEpisodeLabel]` and `$VAR[SeasonEpisodeLabel]` ? I do not understand when you use the one or the other.

emveepee commented:
> 1. Uncommented force-pushes with additional functional changes after approval is not a good approach.
> 
>     2. Why do we have now `$VAR[ListItemSeasonEpisodeLabel]` and `$VAR[SeasonEpisodeLabel]` ? I do not understand when you use the one or the other.

@ksooo sorry, I realized the Home screen was duplicating the episode name in the boxes but it was very hard to notice.  You didn't reply to my original post about squashing the commits, and generally @phunkyfish has asked me to do that, but I will wait for the request in the future.

The ListItem variant was for list items that had the S/E information in grey in the release skin.  I originally named it SeasonEpisodeHighliteLabel but was asked to change that.  Since the release skin uses both styles the purpose of this PR was not to change functionality just simplify the maintainability of the code by using common variables.  I will be happy to incorporate your suggested changes and add additional highlighting labels assuming @jjd-uk agrees.

jjd-uk commented:
I had missed $VAR[SeasonEpisodeLabel] already existed, so perhaps it can be reused to avoid the new VAR and the grey formatting added when needed.

emveepee commented:
> I had missed `$VAR[SeasonEpisodeLabel]` already existed, so perhaps it can be reused to avoid the new VAR `$VAR[ListItemSeasonEpisodeLabel]` and the grey formatting added when needed.

The two variables both combine season episode episodepart and episode name eliminating the prefixed  S E ? and : when necessary.  The difference is the ListItem variable has embedded  [Color grey] around season episode and episode part.  If you add colour to SeasonEpisodeLabel it would have the same colour for the whole string.

jjd-uk commented:
I haven't had chance to have a proper look at the code yet so that was only off the top of my head.

jjd-uk commented:
Following @ksooo comments I think we can do better. Sorry I've had very limited time so had only reviewed what was presented, now I've had chance to take a bit of a deeper dive into the code.

I think we can remove `$VAR[ListItemSeasonEpisodeLabel]` and simply use `$VAR[SeasonEpisodeLabel]` meaning only these changes are necesaary https://gist.github.com/jjd-uk/ed934430217655152ba70621fb195334

I haven't had time to extensively test hence I've removed my review as I'll need to revisit another day to be 100% sure. Let me know if you find any issues.

emveepee commented:
@jjd-uk the current state is closest to master changing to one variable only you have either greying of S/E or none.  The reason for this is that EpisodeName is part of the variable now too, so condition testing of it wasn't necessary to avoid the trailing semi colon. To see how they are closely like in the source try grep SeasonEpisodeLabel *.xml

```
DialogPVRInfo.xml:				<param name="sub_label" value="$VAR[FlagDashLabel][COLOR grey]$VAR[SeasonEpisodeLabel][/COLOR]$INFO[ListItem.EpisodeName,[COLOR white][B],[/B][/COLOR]]" />
Home.xml:							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
Home.xml:							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
Home.xml:							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
Home.xml:							<param name="label2" value="$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]"/>
Includes_PVR.xml:				<label>[I]$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName][/I]</label>
MyPVRGuide.xml:							<label>$VAR[FlagDashLabel][I][COLOR grey]$VAR[SeasonEpisodeLabel][/COLOR]$INFO[ListItem.EpisodeName,[COLOR white],[/COLOR]][/I]</label>
Variables.xml:		<value condition="[!String.IsEmpty(ListItem.Season) | !String.IsEmpty(ListItem.Episode) | !String.IsEmpty(ListItem.EpisodeName)] + Window.IsActive(videoplaylist)">$INFO[ListItem.Title,,: ]$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]</value>
Variables.xml:	<variable name="SeasonEpisodeLabel">
Variables.xml:		<value condition="$EXP[listitem_has_episode_info] + !String.IsEmpty(ListItem.EpgEventTitle) + !String.StartsWith(ListItem.EpisodeName,ListItem.EpgEventTitle)">$INFO[ListItem.EpgEventTitle] | [COLOR grey]$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName][/COLOR]</value>
Variables.xml:		<value condition="$EXP[listitem_has_episode_info]">[COLOR grey]$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName][/COLOR]</value>
Variables.xml:		<value condition="$EXP[listitem_has_episode_info] + !String.IsEmpty(ListItem.EpgEventTitle) + !String.StartsWith(ListItem.EpisodeName,ListItem.EpgEventTitle)">$INFO[ListItem.EpgEventTitle] | $VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]</value>
Variables.xml:		<value condition="$EXP[listitem_has_episode_info]">$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]</value>
```

What is the concern with two variables other that not understanding the reason for the change?

jjd-uk commented:
That's not right as Master already uses $VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]

jjd-uk commented:
Only a formatting change from Master, could be left unchanged https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/xml/DialogPVRInfo.xml#L120

Zero change from Master
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Home.xml#L418
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Home.xml#L429
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Home.xml#L501
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Home.xml#L512
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Includes_PVR.xml#L437
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Includes_PVR.xml#L544
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Variables.xml#L86
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Variables.xml#L657
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Variables.xml#L658
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Variables.xml#L663
https://github.com/xbmc/xbmc/blob/c3c9cd67d045c57f35419822fb9deb3329863d77/addons/skin.estuary/xml/Variables.xml#L664

emveepee commented:
> [EDIT] Perhaps you did not notice I removed your change to add $INFO[ListItem.EpisodeName] to $VAR[SeasonEpisodeLabel]

I did notice, but I felt it made more sense to include EpisodeName since it was always paired, and I was adding EpisodePart to SeasonEpisodLabel anyway.   This removes the current condition checking String.IsEmpty(ListItem.EpisodeName) in SeasonEpisodeLabel by making the semicolon part of the $INFO prefix for EpisodeName making

jjd-uk commented:
Have you tested my code? If so please point where there is a problem.

emveepee commented:
@jjd-uk it feels like to you want to use the first commit.  You will there see all the conditions I needed to address to deal with the separate EpisodeName that are reduced in the second commit to Variables.xml

jjd-uk commented:
To be honest I don't know. I simply used my available time this morning to go through what would be required for a minimal change from Master to take account of ksooo's comments. I tried different combinations of EPG data for example 

![image](https://github.com/user-attachments/assets/7ce73318-0b8b-4166-9a29-b6defa42005e)
![image](https://github.com/user-attachments/assets/5a86a7b3-1fa4-4b93-98da-c8131a4e6ca4)

jjd-uk commented:
I really hope you don't think I'm messing you about. With my currently limited time I've been squeezing in the code review to try and not hold this PR up. Perhaps that was a mistake and I should of waited until I had a bigger block of time available.

emveepee commented:
> To be honest I don't know. I simply used my available time this morning to go through what would be required for a minimal change from Master to take account of ksooo's comments. I tried different combinations of EPG data for example

I disagree with any idea that that a minimal change is best.  I was trying to clarify the role of EpisodeName in the variable and reduce the number of conditions that needed to be checked and use the $INFO prefix and suffix whenever possible. 

I think it is best if I remove Estuary from the PR and let you guys figure out how you want to display it when you have time, like other skinners have to do, you may not think it is even necessary.   My original PR was simply to have the EpisodePart appended to List.Episode string in core when available which would have the least impact on all skins. Even pvr.demo hasn't been looked at so other skins can't easily test this if they wanted.

ksooo commented:
> My original PR was simply to have the EpisodePart appended to List.Episode string in core when available which would have the least impact on all skins.

Yes, and this was wrong from the conceptional point of view. How GUI elements should be combined never is something core should decide. Append/Prevent/Put in brackets/... - all this can be skin and/or language dependent.

emveepee commented:
> > My original PR was simply to have the EpisodePart appended to List.Episode string in core when available which would have the least impact on all skins.
> 
> Yes, and this was wrong from the conceptional point of view. How different GUI elements should be combined never is something core should decide. Append/Prevent/Put in brackets/... - all this can be skin and/or language dependent.

Agreed  which is why I changed it.   However you never answered my redirect on why core is able to force the very localized s#e# in the core LabelFormatter or why Estuary continues that dependency.  Again removing the skin from this PR will let you guys figure out how to internationalize the Episode + "slash"  + EpisodePart" combination in Estuary.

jjd-uk commented:
Perhaps this might better show the changes I'd do to Estuary to keep it minimal https://github.com/jjd-uk/xbmc/commit/4155c36aae1a4cabee0244bee4fa0e16d4d85b21

ksooo commented:
> However you never answered my redirect on why core is able to force the very localized s#e# in the core LabelFormatter or why Estuary continues that dependency.

Sorry, I do not understand these questions without more context.

emveepee commented:
> Perhaps this might better show the changes I'd do to Estuary to keep it minimal [jjd-uk@4155c36](https://github.com/jjd-uk/xbmc/commit/4155c36aae1a4cabee0244bee4fa0e16d4d85b21)

OK rather then removing Estuary I have copied your skin into this PR.  For line 665 a removed a trailing space.

jjd-uk reviewed (approved):
Approved from the skin side once minor error is addressed.

emveepee commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -660,7 +668,7 @@
 		<value condition="ListItem.IsFolder">$INFO[ListItem.Timertype]</value>
-		<value condition="$EXP[listitem_has_episode_info] + !String.IsEmpty(ListItem.EpgEventTitle) + !String.StartsWith(ListItem.EpisodeName,ListItem.EpgEventTitle) + !String.IsEqual(ListItem.EpgEventTitle,ListItem.Label)">$INFO[ListItem.EpgEventTitle] | [COLOR grey]$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName][/COLOR]</value>
+		<value condition="$EXP[listitem_has_episode_info] + !String.IsEmpty(ListItem.EpgEventTitle) + !String.StartsWith(ListItem.EpisodeName,ListItem.EpgEventTitle) + !String.IsEqual(ListItem.EpgEventTitle,ListItem.Label)">$INFO[ListItem.EpgEventTitle] | $VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]</value>
 		<value condition="$EXP[listitem_has_episode_info]">$VAR[SeasonEpisodeLabel]$INFO[ListItem.EpisodeName]</value>
```
OK, sorry I was away on a short vacation.  I updated line 671  with the colouring but did not include the label check from the reverted PR. Hopefully this is ok now.

emveepee commented:
@jjd-uk I noticed a typo in your line 671  (EpisodeNam  not EpisodeName) so i pushed another change.

jjd-uk commented:
@ksooo from my side all looks good with this now.

ksooo commented:
> from my side all looks good with this now.

@jjd-uk tbh, I cannot oversee what unwanted side effects this (imo complex) changes might have. But let's merge and see what happens.

ksooo commented:
jenkins build this please

ksooo commented:
jenkins build this please

emveepee commented:
@ksooo you parked this PR 3 weeks ago.  Should I abandon it or should I update it for the EpisodePart on recordings.

ksooo commented:
@emveepee I lost track a bit, sorry. Could you please post screenshots how it looks like in Estuary with code as it is in this PR?

emveepee commented:
> @emveepee I lost track a bit, sorry. Could you please post screenshots how it looks like in Estuary with code as it is in this PR?

@ksooo here you go https://imgur.com/a/7DnP33I  using the PR 159 I have for pvr.demo.  The video player change is probably the cleanest since it doesn't strip out Episode.  The last screen is from Recordings where the part number is not yet enabled.

I have asked the user for his database I don't have UK OTA data should you need "real" data.

ksooo commented:
@emveepee thanks for the screen shots. I'd say go ahead with this PR and add (please in a separate commit) recordings support.

emveepee commented:
> @emveepee thanks for the screen shots. I'd say go ahead with this PR and add (please in a separate commit) recordings support.

@ksooo once this is merged I will do a recording PR.  I also noted that these also need to be documented on the forum after each merge.

ksooo commented:
jenkins build this please
